c_plus_plus/src/model/stream/stream.cpp

```diff
@@ -85,14 +92,49 @@ void Stream::start()
 {
     mediaThread_->start();
     detectionThread_->start();
+    odasClient_->start();
+
+    status_ = StreamStatus::RUNNING;
+    emit statusChanged();
 }
 
 void Stream::stop()
 {
-    detectionThread_->stop();
-    detectionThread_->join();
-    mediaThread_->stop();
-    mediaThread_->join();
+    status_ = StreamStatus::STOPPING;
+    emit statusChanged();
+
+    if (odasClient_ != nullptr && odasClient_->getState() != OdasClientState::CRASHED)
```

Comment:
Again, all these nullptr check are redundant, if the construction of any of these members fail, the stream object construction will fail